### ISSUE TITLE
Mixer Centrifuge returning a null [1.12.1]

### ISSUE THREAD

**Issue by teagan75** [RootsTeam/Embers#248]
Forge version 14.22.0.2467
Embers version 0.221

When using goggles from Crossroads, every time I look at this machine it returns a null and causes my player to tick. 

https://pastebin.com/5igmghxY

**Comment by HolyDLX** [comment]
Had some spare time.

### Analysis

If you look at a tile entity with the *GoggleLenses* equiped it checks if the tile entity has the *FLUID_HANDLER_CAPABILITY* using *NULL* as the face. If so it gets the fluid handler or whatever from the entity and iterates over contents.

The *TileEntityMixerBottom* states that it has the requested capability if the face is not *UP* nor *DOWN*. But the *GoggleLenses* use *NULL* for the facing resulting in *TileEntityMixerBottom* responding that it has the capability.

If *NULL* facing was supplied in the call of *getCapability* the *TileEntityMixerBottom* leaves the switch case and asks the super class. The super class *TileEntity* returns NULL.

### How To Fix

This can either be fixed by the *GoggleLenses* providing the facing they are looking at or by adjusting *hasCapability*. I guess the best way would be to fix the *hasCapability* implementation of *TileEntityMixerBottom* to prevent future issues with other mods.

### Crossroads Code Reference

https://github.com/Da-Technomancer/Crossroads/blob/708a194be231a70550b663b1969bd0ccdc89ae01/src/main/java/com/Da_Technomancer/crossroads/API/effects/goggles/DiamondGoggleEffect.java#L25

```
@Override
public void armorTick(World world, EntityPlayer player, ArrayList<String> chat, RayTraceResult ray){
	//Notice that null side is used for finding the capability. This is because most blocks with fluid support only have one side with the capability, which would normally be inaccessible. 
	if(ray != null){
		TileEntity te = world.getTileEntity(ray.getBlockPos());
		if(te != null && te.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null)){
			IFluidHandler fluids = te.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null);
			for(IFluidTankProperties tank : fluids.getTankProperties()){
				chat.add("Type: " + (tank.getContents() == null ? "None" : tank.getContents().getLocalizedName()) + ", Amount: " + (tank.getContents() == null ? 0 : tank.getContents().amount) + ", Capacity: " + tank.getCapacity() + ", Pressure: " + ((double) (tank.getContents() == null ? 0 : tank.getContents().amount)) / ((double) tank.getCapacity()));
			}
		}
		if(te instanceof IInfoTE){
			((IInfoTE) te).addInfo(chat, GoggleLenses.DIAMOND, player, ray.sideHit);
		}
	}
}
```

**Comment by elucent** [comment]
Should be fixed in 0.230.
